src/scripts/main.js

```diff
@@ -1,3 +1,27 @@
 'use strict';
 
-// write code here
+document.querySelectorAll('thead th').forEach((th) => {
+  th.addEventListener('click', () => {
+    const cellIndex = th.cellIndex;
+    const tableBody = document.querySelector('table tbody');
+    const rows = Array.from(tableBody.rows);
+
+    rows.sort((row1, row2) => {
+      let val1 = row1.cells[cellIndex].textContent.trim();
+      let val2 = row2.cells[cellIndex].textContent.trim();
+
+      if (
+        !isNaN(val1.replace(/[$,]/g, '')) &&
+        !isNaN(val2.replace(/[$,]/g, ''))) {
+        val1 = parseFloat(val1.replace(/[$,]/g, ''));
+        val2 = parseFloat(val2.replace(/[$,]/g, ''));
+
+        return val1 - val2;
+      }
+
+      return val1.localeCompare(val2);
```

Comment:
The `localeCompare` method is used for string comparison, which is appropriate for handling different character sets and locales. Ensure that the locale settings are suitable for your application's audience.